unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/Editor/Procedural Skybox/SkyboxEditor.cs

```diff
@@ -0,0 +1,1344 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+
+namespace DCL.Skybox
+{
+    public class SkyboxEditorWindow : EditorWindow
+    {
+        private List<SkyboxConfiguration> configurations;
+        private List<string> configurationNames;
+        private int selectedConfigurationIndex;
+        private int newConfigIndex;
+        private SkyboxConfiguration selectedConfiguration;
+
+        public bool isPaused;
+        public float timeOfTheDay;
+        private Material selectedMat;
+        private Vector2 panelScrollPos;
+        private Light directionalLight;
+        private bool creatingNewConfig;
+        private string newConfigName;
+
+        private bool showBackgroundLayer;
+        private bool showAmbienLayer;
+        private bool showFogLayer;
+        private bool showDLLayer;
+
+        public static SkyboxEditorWindow instance { get { return GetWindow<SkyboxEditorWindow>(); } }
+
+        private void OnEnable()
+        {
+            if (selectedConfiguration == null)
+            {
+                EnsureDependencies();
+            }
+
+        }
+
+        void OnFocus()
+        {
+            if (selectedConfiguration == null)
+            {
+                OnEnable();
+            }
+        }
+
+        [MenuItem("Window/Skybox Editor")]
+        static void Init()
+        {
+
+            SkyboxEditorWindow window = (SkyboxEditorWindow)EditorWindow.GetWindow(typeof(SkyboxEditorWindow));
+            //SkyboxEditorWindow.selectedConfiguration = new SkyboxConfiguration();
```

Comment:
```suggestion
```

Reply:
Done. Making all these changes in a new commit